gateway/src/main/java/io/zeebe/gateway/impl/job/LongPollingActivateJobsHandler.java

```diff
@@ -115,42 +138,92 @@ private void unblock(String jobType) {
           LOG.trace("Unblocking ActivateJobsRequest {}", request.getRequest());
           activateJobs(request);
         });
+    state.clearBlockedRequests();
   }
 
-  private boolean isJobAvailable(String jobType) {
-    final Integer notAvailableResponses = zeroResponses.get(jobType);
-    return notAvailableResponses == null || notAvailableResponses < NO_JOBS_RESPONSE_THRESHOLD;
-  }
-
-  private void block(LongPollingActivateJobsRequest request) {
+  private void block(JobTypeAvailabilityState state, LongPollingActivateJobsRequest request) {
     if (!request.isTimedOut()) {
       LOG.trace(
           "Jobs of type {} not available. Blocking request {}",
           request.getType(),
           request.getRequest());
-      blockedRequests.computeIfAbsent(request.getType(), t -> new LinkedList<>()).offer(request);
+      state.blockRequest(request);
       if (!request.hasScheduledTimer()) {
-        addTimeOut(request);
+        addTimeOut(state, request);
       }
     }
   }
 
-  private void addTimeOut(LongPollingActivateJobsRequest request) {
+  private void addTimeOut(JobTypeAvailabilityState state, LongPollingActivateJobsRequest request) {
+    ActorClock.currentTimeMillis();
     final ScheduledTimer timeout =
         actor.runDelayed(
-            DEFAULT_LONG_POLLING_TIMEOUT,
+            longPollingTimeout,
             () -> {
-              try {
-                final Queue<LongPollingActivateJobsRequest> requests =
-                    blockedRequests.get(request.getType());
-                if (requests != null) {
-                  requests.remove(request);
-                }
-                request.timeout();
-              } catch (Exception e) {
-                e.printStackTrace();
-              }
+              state.removeBlockedRequest(request);
+              request.timeout();
             });
     request.setScheduledTimer(timeout);
   }
+
+  private void probe() {
+    final long now = currentTimeMillis();
+    jobTypeState.forEach(
+        (type, state) -> {
+          if (state.getLastUpdatedTime() < (now - probeTimeoutMillis)) {
+            final LongPollingActivateJobsRequest probeRequest = state.pollBlockedRequests();
+            if (probeRequest != null) {
+              activateJobsUnchecked(probeRequest);
+            } else {
+              // there are no blocked requests, so use next request as probe
+              if (state.getEmptyResponses() >= emptyResponseThreshold) {
+                state.resetEmptyResponses(emptyResponseThreshold - 1);
+              }
+            }
+          }
+        });
+  }
+
+  public static Builder newLongPollingActivateJobsHandler() {
```

Comment:
It is more common to call the builder method `LongPollingActivateJobsHandler.newBuilder()` or creating a new instance of the builder `new LongPollingActivateJobsHandler.Builder()`.

Reply:
:+1: